pinot-spi/src/main/java/org/apache/pinot/spi/config/table/SegmentsValidationAndRetentionConfig.java

```diff
@@ -49,9 +49,13 @@ public class SegmentsValidationAndRetentionConfig extends BaseJsonConfig {
   // For more usage of this field, please refer to this design doc: https://tinyurl.com/f63ru4sb
   private String _peerSegmentDownloadScheme;
 
+  // Indicates if the segment should be uploaded to the deep store's file system or to the controller during the
+  // segment commit protocol. By default, segment is uploaded to the controller during commit.
+  // If this flag is set to true, the segment is uploaded to deep store.
+  private boolean _uploadToFileSystem = false;
```

Comment:
I think this flag belongs to the stream ingestion config, suggest moving it to `StreamIngestionConfig`.
For clarity, also suggest renaming to `_serverUploadToDeepStore` because we want to emphasize that we want the server to do the upload

Reply:
> For clarity, also suggest renaming to _serverUploadToDeepStore because we want to emphasize that we want the server to do the upload

This makes sense.

> this flag belongs to the stream ingestion config, suggest moving it to StreamIngestionConfig

yeah. I also agree it fits better in stream ingestion config. Should peer segment download scheme remain in `SegmentsValidationAndRetentionConfig` ?